### DISCUSSION TITLE
MKL error during analysis task on Jet

### DISCUSSION THREAD

**Discussion by mrinalbiswas** [hafs-community/HAFS#194] [General]
I encountered MKL errors when running the analysis task with feature/hafsv1_merge branch on Jet. The error message is copied below.  
I am opening a Discussion thread because few people have encountered this as of now. Biju provided a fix which worked for me. Just wanted to discuss whether this should be included in the repository. 

**_ERROR:_** 
+ 4 + module load gsi_jet
++ 4 + /apps/lmod/8.5.2/libexec/lmod bash load gsi_jet
Lmod has detected the following error: The following module(s) are unknown:
"mkl"
Please check the spelling or version number. Also try "module spider ..."
It is also possible your cache file is out-of-date; it may help to try:
  $ module --ignore_cache load "mkl"
Also make sure that all modulefiles written in TCL start with the string
#%Module

**_FIX:_** 

> diff --git a/jobs/JHAFS_ANALYSIS b/jobs/JHAFS_ANALYSIS
index c1651ca1..88db67ae 100755
--- a/jobs/JHAFS_ANALYSIS
+++ b/jobs/JHAFS_ANALYSIS
@@ -37,6 +37,7 @@ if [ "${RUN_ENVIR^^}" != "NCO" ]; then
   if [ ${machine}  = "hera" ]; then
     module load gsi_${machine}.intel
   else
+    module purge
     module load gsi_${machine}
   fi
   module list

**Comment by BinLiu-NOAA** [comment]
This issue has been fixed and is obsolete now. Closing this discussion.
